### PR TITLE
Changes for building the config-gen tool in mac

### PR DESCRIPTION
**Pull request by bandatarunkumar**
* **Problem Overview**  
To support the internal tool in mac native environment, we need to include new directory wrt to mac and it should only hit only in case of mac compilation, so added the necessary changes under a flag. The changes in this PR will not have any effect when we compile the code in linux environment.
* **Testing Done**  
Tested the changes by generating the configs manually.

### CODE REVIEW

File: threshsign/src/ThresholdAccumulatorBase.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ void ThresholdAccumulatorBase<VerificationKey, NumType, SigShareParserFunc>::set
   assertNotNull(msg);
   assertStrictlyPositive(len);
```
Comment (cjn) [inline threshsign/src/ThresholdAccumulatorBase.cpp:25]: Please add comments in code

File: threshsign/src/bls/relic/BlsAccumulatorBase.cpp [hunk 2]
```diff
@@ -55,8 +55,11 @@ BlsAccumulatorBase::BlsAccumulatorBase(const std::vector<BlsPublicKey>& verifKey
 void BlsAccumulatorBase::onExpectedDigestSet() {
   assertNotNull(expectedDigest);
   assertStrictlyPositive(expectedDigestLen);
-
+#ifdef BUILD_CONFIG_GEN_TOOL_FOR_MAC
```
Comment (cjn) [inline threshsign/src/bls/relic/BlsAccumulatorBase.cpp:58]: Please add comments in code

File: logging/CMakeLists.txt [hunk 3]
```diff
@@ -20,6 +24,7 @@ else(USE_LOG4CPP)
 	target_sources(logging PRIVATE src/Logging.cpp)
 endif(USE_LOG4CPP)
 
-
 set_property(DIRECTORY ${CMAKE_SOURCE_DIR} APPEND PROPERTY LINK_LIBRARIES logging)
```
Comment (cjn) [inline logging/CMakeLists.txt:27]: Please add comment if required

**Comment by cjn** [comment]
LGTM. I have updated few minor comments.

File: threshsign/src/ThresholdAccumulatorBase.cpp (hunk 1 above)
Comment (bandatarunkumar) [reply threshsign/src/ThresholdAccumulatorBase.cpp:25]: this code will not be hit while generating the binary in mac

File: threshsign/src/bls/relic/BlsAccumulatorBase.cpp (hunk 2 above)
Comment (bandatarunkumar) [reply threshsign/src/bls/relic/BlsAccumulatorBase.cpp:62]: as discussed, not adding comment here

File: logging/CMakeLists.txt (hunk 3 above)
Comment (bandatarunkumar) [reply logging/CMakeLists.txt:27]: as discussed, not adding comment here

**Review by cjn — APPROVED** [review]

**Review by toly-kournik — APPROVED** [review]
LGTM

File: util/include/Metrics.hpp [hunk 4]
```diff
@@ -21,6 +21,7 @@
 #include <memory>
 #include <list>
 #include <variant>
+#include <unordered_map>
```
Comment (pkthapa) [inline util/include/Metrics.hpp:24]: nit: why included?

Comment (bandatarunkumar) [reply util/include/Metrics.hpp:24]: in mac.. we specially compile specific folders, due to which the unordered_map include looks missing in that flow, i had to add it to reduce the compilation error..

**Pull request merged**